plugins/modules/fusion_ra.py

```diff
@@ -131,16 +140,14 @@ def get_ra(module, fusion):
     """Return Role Assignment or None"""
     ra_api_instance = purefusion.RoleAssignmentsApi(fusion)
     try:
+        principal = get_principal(module, fusion)
         assignments = ra_api_instance.list_role_assignments(
-            role_name=module.params["role"]
+            role_name=module.params["role"],
+            principal=principal,
         )
         for assign in range(0, len(assignments)):
-            principal = human_to_principal(fusion, module.params["user"])
-            scope = human_to_scope(module.params)
-            if (
-                assignments[assign].principal == principal
-                and assignments[assign].scope.self_link == scope
-            ):
+            scope = get_scope(module.params)
+            if assignments[assign].scope.self_link == scope:
                 return assignments[assign]
```

Comment:
I know it's not your code, but this PR seems like a good place to address this. Can you please replace `for assign in range(0, len(assignments))` by `for assign in assignments`? It's much simpler and removes extra complexity :D

Reply:
Sure, good call. Fixed in [fc973ce](https://github.com/Pure-Storage-Ansible/Fusion-Collection/pull/135/commits/fc973cea2dac05fa3d8f761c50b6695b3db86675).